genomepy/annotation.py

```diff
@@ -714,6 +709,60 @@ def map_locations(self, annot: Union[str, pd.DataFrame], to: str) -> pd.DataFram
         return df
 
 
+@memory.cache
+def query_mygene(
+    query: Iterable[str],
+    tax_id: str,
+    fields: str = "genomic_pos",
+    batch_size: int = 10000,
+) -> pd.DataFrame:
+    """
+    Use mygene.info to map gene identifiers to another type.
+
+    Parameters
+    ----------
+    query: iterable
+        a list or list-like of gene identifiers
+    fields : str, optional
+        Target identifier to map the query genes to. Valid fields
+        are: ensembl.gene, entrezgene, symbol, name, refseq, entrezgene. Note that
+        refseq will return the protein refseq_id by default, use `product="rna"` to
+        return the RNA refseq_id. Currently, mapping to Ensembl transcript ids is
+        not supported.
+    batch_size: int, optional
+        Controls batch size for REST API.
+
+    Returns
+    -------
+    pandas.DataFrame with mapped gene annotation.
+    """
+    logger.info("Querying mygene.info...")
+    query = [_ for _ in query]
```

Comment:
`list(query)`?

Reply:
also, converting large generators into lists isn't very nice. Is there a nice way of taking (up to) `batch_size` elements from the Iterator? How about:
```
def query(batch : list, result : pd.DataFrame) -> pd.DataFrame:
    mg = mygene.MyGeneInfo()
    _result = mg.querymany(...)
    result = pd.concat((result, _result))
    return result

batch = []
result = pd.DataFrame()
for n, q in tqdm(enumerate(query)):  # tqdm had some system to combine it with enumerate...
    batch.append(q)
    if (n+1/batch_size).is_integer():
        result = query(batch, result)
        batch = []

if len(batch) > 0:
    result = query(batch, result)
```
The `tqdm()` bar wont show the total queries, but it will save time with huge generators. Not sure if this is the fastest possible way though.

Reply:
You're correct, but in this case, do you think it matters? I would think the lists to convert will never be larger than  (roughly)10-100,000's. And in practice, these will already mostly be lists/arrays anyway. Also not sure if the multiple `append()`'s will in practice not be slower?

Reply:
Maybe this is indeed overzealous. The query input are gene names, not motif scans :p